clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -270,12 +262,12 @@ public boolean partitionsAutoAssigned() {
         return !this.subscription.isEmpty();
     }
 
-    public void consumed(TopicPartition tp, long offset) {
-        assignedState(tp).consumed(offset);
+    public void position(TopicPartition tp, long offset) {
```

Comment:
Another thing to consider is memory limit on the consumer (we currently do not have such management as we did in producer, but it is tracked in KAFKA-2045). I agree that pre-fetch would be helpful with bursty network, or more generally speaking you would probably want to get some data in each selector.poll() even if there is data buffered for all partitions, as long as buffered data does not exceed the memory limit.